contracts/gateway/OSTPrime.sol

```diff
@@ -225,21 +226,33 @@ contract OSTPrime is UtilityToken, OSTPrimeConfig {
      *      is added in function increaseSupplyInternal.
      *
      * @param _account Account address for which the OST Prime balance will be
-     *                 increased.
+     *                 increased. This is payable so that base token can be
+     *                 transferred to the account.
      * @param _amount Amount of tokens.
      *
      * @return success_ `true` if increase supply is successful, false otherwise.
      */
     function increaseSupply(
-        address _account,
+        address payable _account,
         uint256 _amount
     )
         external
         onlyInitialized
         onlyCoGateway
         returns (bool success_)
     {
-        success_ = increaseSupplyInternal(_account, _amount);
+        /*
+         * Acquire lock for msg.sender so that this function can only be
+         * executed once in a transaction.
+         */
+
+        acquire(msg.sender);
```

Comment:
Not sure, but shouldn't the lock be on `_account`? What if I somehow manage to call this from another contract? 🤔
Then what if I create multiple accounts and contracts and each sends to the next? 🤔

Reply:
It can't be called from another account because it's `onlyCoGateway`. If we lock for coGateway, there is no way to enter.

Reply:
But that means we don't need the mutex in the first place, right? 🤯
Anyways, this works 👍